Longest_Common_Subsequence/Longest_Common_Subsequence.rb

```diff
@@ -0,0 +1,54 @@
+=begin 
+Problem Statement- Given two sequences, find 
+the length of longest subsequence present in both of them. 
+A subsequence is a sequence that can be derived 
+from another sequence by deleting some or 
+no elements without changing the order of the remaining elements.
+=end
+
+def longest_common_subsequence(arr1, arr2, size1, size2) 
+    lcs = Array.new(size1 + 1){Array.new(size2 + 1)}
+    
+    for i in 0..size1
+        for j in 0..size2
+            if (i == 0 || j == 0) 
+                lcs[i][j] = 0    
+            # when the last character of both subsequences match, 
+            # increase length of lcs by 1
+            elsif (arr1[i - 1] == arr2[j - 1]) 
+                lcs[i][j] = lcs[i - 1][j - 1] + 1
+            # when the last character is not same, 
+            # take maximum obtained by adding
+            # one character to one of the subsequences 
+            else
+                lcs[i][j] = [lcs[i - 1][j], lcs[i][j - 1]].max
+            end
+        end 
+    end
+    
+    lcs[size1][size2]
+end 
+
+# Driver Code
+puts "Enter the size of first array:"
+size1 = gets.to_i
+puts "Enter the first array elements:"
+arr1 = gets.split(" ").map(&:to_i)
+puts "Enter the size of second array:"
+size2 = gets.to_i
+puts "Enter the second array elements:"
```

Comment:
Please add one blank line before this for readability.

Reply:
Done